src/main/java/core/basesyntax/service/impl/CsvReportServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.service.impl;
+
+import com.opencsv.CSVWriter;
+import core.basesyntax.service.ReportService;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
+
+public class CsvReportServiceImpl implements ReportService {
+    @Override
+    public void generateReport(Map<String, Integer> fruits, String fileName) {
+        File file = new File(fileName);
+        try {
+            FileWriter outputfile = new FileWriter(file);
+            CSVWriter writer = new CSVWriter(outputfile);
+            String[] header = {"fruit", "quantity"};
```

Comment:
create a constant for HEADER, don't need to create an array of string to write to file, form report using StringBuilder and return one String